src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -783,10 +783,13 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
     try {
       return fn();
     } finally {
-      shouldBatchUpdates = prev;
-      // If we've exited the batch, perform any scheduled task work
-      if (!shouldBatchUpdates) {
-        performTaskWork();
+      // If we're exiting the batch, perform any scheduled task work
+      try {
+        if (!prev) {
+          performTaskWork();
```

Comment:
If more work gets scheduled during performTaskWork, e.g. in componentDidMount, does that still get performed before returning here?